File: Javascript/SortingAlgorithms/BubbleSort.js
// javascript program to implement bubblesort

function bubbleSort(arr) {
    let n = arr.length

    let swapped = false
    for (let i=0; i < n; i++) {
        
        for (let j=0; j<n-i-1; j++) {

            if (arr[j] > arr[j+1]) {
                let temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
                swapped = true
            }
            
        }
        if (swapped == false) break
    }
       
}

function printArray(arr) {

    for (val of arr) {
        process.stdout.write(val + " ")
    }
}

let arr = [64, 34, 25, 12, 22, 11, 90]

bubbleSort(arr)

printArray(arr)

File: Python/SortingAlgorithms/BubbleSort.py
# a python program to implement bubblesort
def bubbleSort(arr):
    n = len(arr)
    
    # Traverse through all array elements
    for i in range(n - 1):
        
        # Last i elements are already in place
        for j in range(n-i-1):
            
            # traverse the array from 0 to n-i-1
            # Swap if the element found is greater
            # than the next element
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
        

# Driver code to test above
arr = [64, 34, 25, 12, 22, 11, 90]

bubbleSort(arr)
print(arr)
    

                